Production deployment

uvNikita opened the pull request.

uvNikita commented:
Ok, now I tested everything from scratch and it works.
All you have to do is install git, clone repository, checkout and run `setup_production` script.

Gachapen commented on README.md:
```diff
@@ -40,3 +40,19 @@ use http://localhost:5000/ in you browser to see the application.
     $ vagrant provision
+
+
+## Production deployment
```
Maybe add Ubuntu as a requirement for this setup? Others should work, but requires some tinkering with the setup I think.

Gachapen commented:
Do you have a deployed version accessible on the Internet?

Gachapen commented:
The current "Installation" and "Run" sections in the readme might be confusing a person who wants to just deploy a production version. I think it would be better if they were renamed to something like "Installation for development" and "Run development version".

Gachapen commented on README.md:
```diff
@@ -40,3 +40,19 @@ use http://localhost:5000/ in you browser to see the application.
+
+Checkout specific version:
+
+    $ git checkout v0.4.0
```
Maybe <version> in stead of v0.4.0?

uvNikita replied:
What do you mean?

Gachapen replied:
Oups, markdown didn't like the \<\> characters.

Gachapen commented:
I tried setting it up on an Ubuntu 15.10 machine (raw install from CD image) and `npm install` fails with
```
TASK: [install dependencies with npm] ***************************************** 
failed: [localhost] => {"changed": true, "cmd": ["/usr/bin/npm", "install", "--silent", "--no-bin-links"], "delta": "0:00:00.970616", "end": "2015-11-21 13:38:26.290363", "rc": 34, "start": "2015-11-21 13:38:25.319747", "warnings": []}

FATAL: all hosts have already failed -- aborting
```

Gachapen commented:
The issue was that I used a relative path to `/path/to/sources`. Absolute path works. You should either explain in the docs that it has to be an absolute path, or somehow allow for relative paths (you should be able to get the absolute from the relative with bash).

Gachapen commented:
Now supervisor has issues starting:
```
TASK: [start supervisor] ****************************************************** 
failed: [localhost] => {"failed": true}
msg: Job for supervisor.service failed because the control process exited with error code. See "systemctl status supervisor.service" and "journalctl -xe" for details.
```
```
● supervisor.service - LSB: Start/stop supervisor
   Loaded: loaded (/etc/init.d/supervisor)
   Active: failed (Result: exit-code) since lø. 2015-11-21 13:51:11 CET; 1min 10s ago
     Docs: man:systemd-sysv-generator(8)
  Process: 15636 ExecStop=/etc/init.d/supervisor stop (code=exited, status=0/SUCCESS)
  Process: 15658 ExecStart=/etc/init.d/supervisor start (code=exited, status=1/FAILURE)

nov. 21 13:51:11 magnus-VirtualBox systemd[1]: Starting LSB: Start/stop supervisor...
nov. 21 13:51:11 magnus-VirtualBox supervisor[15658]: Starting supervisor:
nov. 21 13:51:11 magnus-VirtualBox systemd[1]: supervisor.service: Control process exited, code=exited status=1
nov. 21 13:51:11 magnus-VirtualBox systemd[1]: Failed to start LSB: Start/stop supervisor.
nov. 21 13:51:11 magnus-VirtualBox systemd[1]: supervisor.service: Unit entered failed state.
nov. 21 13:51:11 magnus-VirtualBox systemd[1]: supervisor.service: Failed with result 'exit-code'.
```

uvNikita commented:
The last machine is available here: http://162.243.223.61/student

uvNikita commented:
I rewrote README file and made it possible to specify relative path in script. Also fixed problems with `supervisor` service. For some reasons I didn't experience those errors on virtual machine, but had them locally. Now everything should work correctly.
Tested from scratch on Ubuntu 15.10 (http://104.131.213.213/student/).

Gachapen commented:
`ansible-playbook` works perfectly now, and the changes in the readme looks good. But I still had some problems. I got 404 from nginx on all URLs (e.g. localhost/student/), but fixed it by restarting nginx. It looks like the ansible config should do it, but it doesn't seem like it actually is doing it.

Gachapen commented:
The lecturer engagement view isn't showing and I get this error in the console log:
```
Error: [$injector:unpr] Unknown provider: CONSTANTProvider <- CONSTANT <- lecturesService <- lecturesFactory
http://errors.angularjs.org/1.4.8/$injector/unpr?p0=CONSTANTProvider%20%3C-%20CONSTANT%20%3C-%20lecturesService%20%3C-%20lecturesFactory
    at http://104.131.213.213/release/min/IFS-deps.min.js:4:25243
    at http://104.131.213.213/release/min/IFS-deps.min.js:5:12873
    at Object.d [as get] (http://104.131.213.213/release/min/IFS-deps.min.js:5:12043)
    at http://104.131.213.213/release/min/IFS-deps.min.js:5:12970
    at d (http://104.131.213.213/release/min/IFS-deps.min.js:5:12043)
    at e (http://104.131.213.213/release/min/IFS-deps.min.js:5:12380)
    at Object.f [as instantiate] (http://104.131.213.213/release/min/IFS-deps.min.js:5:12505)
    at Object.<anonymous> (http://104.131.213.213/release/min/IFS-deps.min.js:5:11033)
    at Object.e [as invoke] (http://104.131.213.213/release/min/IFS-deps.min.js:5:12413)
    at Object.$get (http://104.131.213.213/release/min/IFS-deps.min.js:5:10803) <div data-ng-view="" class="ng-scope">(anonymous function) @ IFS-deps.min.js:6
lecturer.min.js:1 Route change (Lecture Controller)
lecturer.min.js:1 Route change (Header Controller)
```
I think there is some issue with the release build.

121052 commented:
I believe this is fixed on the IFS-101 (in progress) front end lecturer login.

uvNikita commented:
Replaced 'v0.4.0' string.
As for `nginx`, I don't know why it didn't restarted for you, but for me it restarts each time `nginx_site_dev.conf` file changes.

Gachapen commented:
I successfully tested it now. I also tested setting up MySQL and using phpmyadmin to modify the database. It worked perfectly after figuring out some issues. Please see my commits. If they're OK, I'll merge this PR.

uvNikita commented:
Awesome
